File: deepsense/sandbox/runner.py
"""
Python runner for sandbox execution.
"""

import subprocess
import sys
import json

with open("script/meta.json") as f:
    meta = json.load(f)

code = meta.get("code", "")
reqs = meta.get("requirements", [])

# Install packages with suppressed warnings
if reqs:
    subprocess.run([
        sys.executable, "-m", "pip", "install", 
        "--quiet", "--no-warn-script-location", "--disable-pip-version-check"
    ] + reqs, capture_output=True)

# Write code to file
with open("script/user_script.py", "w") as f:
    f.write(code)

# Execute code
result = subprocess.run(["python", "script/user_script.py"], capture_output=True, text=True)
print(result.stdout)
print(result.stderr, file=sys.stderr)

File: deepsense/sandbox/runner.js
const { execSync } = require('child_process');
const fs = require('fs');

const meta = JSON.parse(fs.readFileSync('script/meta.json'));
const code = meta.code || '';
const packages = meta.requirements || [];

// Install npm packages with suppressed warnings
if (packages.length > 0) {
  try {
    execSync(`npm install --silent --no-audit --no-fund ${packages.join(' ')}`, { 
      stdio: 'pipe',
      encoding: 'utf-8'
    });
  } catch (err) {
    console.error('NPM install failed:', err.message);
  }
}

// Write code to file
fs.writeFileSync('script/user_script.js', code);

// Execute code
try {
  const output = execSync('node script/user_script.js', { encoding: 'utf-8' });
  console.log(output);
} catch (err) {
  console.error('Execution error:', err.stderr || err.message);
}

